### DISCUSSION TITLE
Question on 'display_history' function

### DISCUSSION THREAD

**Discussion by FieClaussell123** [eleurent/highway-env#167] [General]
Hi !  I met some problems when I wanted to use  [display_history](https://github.com/eleurent/highway-env/blob/6b4ad85462a33ebfa1b7d1de43cfbe85a59bd77e/highway_env/vehicle/graphics.py#L119) function. 

I used `env.configure({"show_trajectories": True})` to implement it. I'm sure it can be use in current version of highway-env. 
But i dont know why the error happened on my version. I used the env since 3 months ago.
I know there are some differents in the code between that time and now.

I want to know if the function cant be used in that old version (3 months ago) or I should use other environment configurations ?

**Comment by mhtb32** [comment]
It would be helpful if you mention the latest commit in the git history of your current version of highway-env.

**Comment by FieClaussell123** [comment]
Sorry, I can only know that I downloaded the highway-env on 29 March.

**Comment by eleurent** [comment]
I tried it on the commit 23d159736660f7b8e4c2cf261781d73c60437e84 of March 29, and it works properly. The feature dates back to around October 2019.

What error do you get, exactly?

**Comment by FieClaussell123** [comment]
The error I got like this :

```
highway_env/vehicle/dynamics.py in create_from(cls, vehicle)
     95         :return: a new vehicle at the same dynamical state
     96         """
---> 97         v = cls(vehicle.road, vehicle.position, vehicle.heading, vehicle.velocity)
     98         return v
     99 

TypeError: __init__() takes from 3 to 4 positional arguments but 5 were given
```

I only post the latest graph.

**Comment by FieClaussell123** [comment]
```
---> 97         v = cls(vehicle.road, vehicle.position, vehicle.heading, vehicle.velocity)
```

i delete the vehicle.velocity like this : 

```
v = cls(vehicle.road, vehicle.position, vehicle.heading)

```

then it works. Is the vehicle.velocity needed ?

**Comment by eleurent** [comment]
I wonder what is going on, all Vehicle classes have required (at least) 4 arguments for very a long time...
You can remove vehicle.velocity if it seems to be working fine, but it might have hidden unintended consequences. I would rather advise to uninstall highway-env and reinstall the latest version.

**Comment by FieClaussell123** [comment]
yep ! i agree, thanks for your reply . I will quit current version and reinstall the latest version haha.
